src/React.Owin/JsxFileMiddleware.cs

```diff
@@ -0,0 +1,71 @@
+﻿/*
+ *  Copyright (c) 2014-2015, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant 
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+using Microsoft.Owin.StaticFiles;
+
+namespace React.Owin
+{
+	/// <summary>
+	/// Enables serving static JSX files transformed to pure JavaScript. Wraps around StaticFileMiddleware.
+	/// </summary>
+	public class JsxFileMiddleware
+	{
+		private readonly StaticFileMiddleware _internalStaticMiddleware;
+
+		static JsxFileMiddleware()
+		{
+			Initializer.Initialize(_ => _);
```

Comment:
Owin doesn't use any IoC container - it just doesn't have to. Owin is an abstraction on hosting container (like IIS, or self hosting) and you can use whatever container suits you (you have a control of all your objects lifetime)

Owin also doesn't have any global / static variable that would let you differentiate between the requests. From middleware perspective, the requests begins with the Invoke methods, and ends with it (asynchronously). Since it can use different threads internally, you can't differentiate the requests by thread id.

All of this makes it hard to create "lifetime provider" similar to ASP.NET version (where you have the global HttpContext.Current).

My fix to this is as follows: 
 1. I set all "per request" objects to be created everytime they are resolved. (https://github.com/mandrek44/React.NET/blob/react.owin/src/React.Owin/JsxFileMiddleware.cs#L31)
 2. When beginning the request I explicitly resolve the only object I need - that's the IReactEnvironment implementation. That's the only place it's requested. (https://github.com/mandrek44/React.NET/blob/react.owin/src/React.Owin/JsxFileMiddleware.cs#L60)
 3. When the request is completed, I explicitly dispose the object.
(https://github.com/mandrek44/React.NET/blob/react.owin/src/React.Owin/JsxFileMiddleware.cs#L66)

This will work as long as nobody starts using the IoC container as Service Locator inside React core library.